src/dra/7E4BC.c

```diff
@@ -1217,7 +1217,260 @@ void EntityPlayerPinkEffect(Entity* self) {
 }
 
 // player dissolves into pixels
-INCLUDE_ASM("dra/nonmatchings/7E4BC", EntityPlayerDissolves);
+void EntityPlayerDissolves(Entity* self) {
+    const int PrimCount = 36;
+    const int Iterations = 40;
+    u8 sp4F;
+    u8 sp4E;
+    u8 sp4D;
+    u8 sp4C;
+    s16 sp4A;
+    s16 sp48;
+    s16 sp46;
+    s16 sp44;
+    s16 sp42;
+    s32 sp3C;
+    s16* sp38;
+    Primitive* prim;
+    s32 temp_i;
+    u8* s2;
+    s16 s3;
+    s16 s4;
+    s16 s5;
+    s16 s6;
+    s16 s7;
+    u32* data;
+    u8* temp_v1;
+    u8* temp1;
+    u8* temp2;
+
+    if (PLAYER.ext.generic.unkAC != 0x38 && PLAYER.ext.generic.unkAC != 0x39 &&
+        PLAYER.ext.generic.unkAC != 0x3A &&
+        (self->step == 2 || self->step == 3)) {
+        self->step = 4;
+        LoadImage(&D_800AE138, D_80139A7C);
+        self->ext.dissolve.unk7E = rand() & 0xFF;
+        self->ext.dissolve.unk7C = 0x30;
+        prim = &g_PrimBuf[self->primIndex];
+        for (temp_i = 0; temp_i < PrimCount; temp_i++) {
+            prim->drawMode &= ~DRAW_HIDE;
+            prim = prim->next;
+        }
+    }
+
+    s5 = self->posX.i.hi = PLAYER.posX.i.hi;
+    s7 = self->posY.i.hi = PLAYER.posY.i.hi;
+    self->facingLeft = PLAYER.facingLeft;
+    if (!(PLAYER.animCurFrame & 0x7FFF)) {
+        DestroyEntity(self);
+        return;
+    }
+
+    sp38 = D_800CF324[PLAYER.animCurFrame & 0x7FFF];
+#ifdef VER_PSP
+    sp42 = (sp38++)[0];
+#else
+    sp42 = sp38[0];
+#endif
+    sp42 &= 0x7FFF;
+    temp_v1 = ((u8**)SPRITESHEET_PTR)[sp42];
+    sp4F = 4;
+    sp4E = 1;
+    sp4D = (sp4F + temp_v1[0]);
+    sp4C = (sp4E + temp_v1[1]);
+    sp46 = (sp4D - sp4F);
+    sp44 = (sp4C - sp4E);
+    s3 = (sp46 / 6);
+    s4 = (sp44 / 6);
+#ifndef VER_PSP
+    sp38++;
+#endif
+    sp4A = ((s16*)sp38)[0] + temp_v1[2];
+    sp48 = ((s16*)sp38)[1] + temp_v1[3];
```

Comment:
```suggestion
    sp4A = sp38[0] + temp_v1[2];
    sp48 = sp38[1] + temp_v1[3];
```
`sp38` is already `s16*`, so you do not need to do a cast.

`temp_v1[2]` and `temp_v1[3]` are `xPivot` and `yPivot` in the context of a player spritesheet (`tools/splat_ext/spritesheet.py` line `214`), so I think you can end up renaming `s5` as `xPivot` and `s7` as `yPivot`